tls/s2n_client_finished.c

```diff
@@ -26,11 +26,13 @@
 #include "utils/s2n_safety.h"
 
 S2N_RESULT s2n_finished_recv(struct s2n_connection *conn, uint8_t *our_version);
-S2N_RESULT s2n_finished_send(struct s2n_connection *conn, uint8_t *seq_num, uint8_t *our_version);
+S2N_RESULT s2n_finished_send(struct s2n_connection *conn, uint8_t *our_version);
 
 int s2n_client_finished_recv(struct s2n_connection *conn)
 {
     POSIX_ENSURE_REF(conn);
+
+    POSIX_GUARD_RESULT(s2n_prf_client_finished(conn));
```

Comment:
I think we actually could always calculate the verify_data at the point. The CCS message is never added to our handshake hash because it's not a handshake message, so for the non-npn path there should be no difference.

The only downside is that we lose the implicit "CCS message must be received before the Finished message" check. Technically that is enforced by the handshake arrays anyway, so losing it wouldn't be the end of the world and I'm not sure keeping it is worth the complication. We could also double check that the handshake array tests have a "ccs message must always appear before the Finished message" check.

Reply:
Did you verify that the tests enforce the "CCS before Finished" rule?